crates/bevy_transform/src/systems.rs

```diff
@@ -323,47 +323,4 @@ mod test {
             assert_eq!(global, &GlobalTransform::from_translation(translation));
         }
     }
-
-    #[test]
-    #[should_panic]
-    fn panic_when_hierarchy_cycle() {
```

Comment:
I removed this test because it should not be possible to trigger a hierarchy cycle outside of `bevy_hierarchy` anymore.